File: DSA /Unit 05/Lec 33/Q2.cpp
Single File Programming Question
Problem Statement



Shafir is working on implementing a hash table with linear probing. He is tasked with inserting several keys into the hash table and then printing the final state of the table. The table should display the inserted keys and the empty slots after all insertions.



Help him implement the task.

Input format :
The first line of input consists of an integer n, representing the number of keys to insert into the hash table.

The second line of input consists of n integers, representing the keys to be inserted into the hash table

Output format :
For each index of the hash table, if a key is present, the output prints: "Index <i>: <key>"

If an index is empty, the output prints: "Index <i>: Empty"



Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

hash table size = 7

1 ≤ n ≤ 100

-106 ≤ key ≤ 106

Sample test cases :
Input 1 :
5
10 20 30 40 50
Output 1 :
Index 0: Empty
Index 1: 50
Index 2: 30
Index 3: 10
Index 4: Empty
Index 5: 40
Index 6: 20
Input 2 :
6
10 20 30 40 50 60
Output 2 :
Index 0: Empty
Index 1: 50
Index 2: 30
Index 3: 10
Index 4: 60
Index 5: 40
Index 6: 20




#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;

    const int TABLE_SIZE = 7;
    vector<long long> table(TABLE_SIZE, LLONG_MIN);

    for (int i = 0; i < n; i++) {
        long long key;
        cin >> key;

        int index = ((key % TABLE_SIZE) + TABLE_SIZE) % TABLE_SIZE;

        while (table[index] != LLONG_MIN) {
            index = (index + 1) % TABLE_SIZE;
        }
        table[index] = key;
    }

    for (int i = 0; i < TABLE_SIZE; i++) {
        if (table[i] == LLONG_MIN)
            cout << "Index " << i << ": Empty\n";
        else
            cout << "Index " << i << ": " << table[i] << "\n";
    }

    return 0;
}

File: Java/Unit 03/CRPC/Q2.java
Single File Programming Question
Problem Statement



Ashok is writing a program that focuses on calculating and displaying the calories burned during a workout for a fitness enthusiast. He has certain specifications as follows.



Implement a class structure with a base class Person and a derived class FitnessEnthusiast that inherits from Person. The program should take user input for the fitness enthusiast's name, workout duration (in minutes), and calorie burn rate (calories burned per minute).



Create a constructor for the FitnessEnthusiast class to initialize the name, workout duration, and calorie burn rate. Utilize the super keyword to invoke the constructor of the superclass. 



Implement a method that calculates the calories burned during the workout using the formula (workout duration * calories burn rate). Display the calculated calories burned.



Help Ashok to complete the program. 

Input format :
The first line of input consists of a string, representing the name.

The second line consists of an integer, representing the workout duration in minutes.

The third line consists of a double value, representing the calorie burn rate.

Output format :
The output prints "Calories Burned: X calories", where X is a double value, representing the calories burned, rounded off to one decimal place.



Refer to the sample output for formatting specifications.

Code constraints :
The given testcases fall under the following constraints:

1 ≤ workout duration ≤ 1200

1.0 ≤ calorie burn rate ≤ 50.0

Sample test cases :
Input 1 :
Dinesh
30
9.70
Output 1 :
Calories Burned: 291.0 calories
Input 2 :
Sharon
120
28.14
Output 2 :
Calories Burned: 3376.8 calories
Whitelist
Set 1:
super



Import java.util.Scanner
// Base class
class Person {

    String name;

    // Constructor of Person
    Person(String name) {
        this.name = name;
    }
}

// Derived class
class FitnessEnthusiast extends Person {

    int duration;
    double burnRate;

    // Constructor using super keyword
    FitnessEnthusiast(String name, int duration, double burnRate) {
        super(name); // calling parent constructor
        this.duration = duration;
        this.burnRate = burnRate;
    }

    // Method to calculate calories
    double calculateCalories() {
        return duration * burnRate;
    }
}
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String name = sc.nextLine();
        int duration = sc.nextInt();
        double burnRate = sc.nextDouble();

        FitnessEnthusiast f = new FitnessEnthusiast(name, duration, burnRate);
        System.out.printf("Calories Burned: %.1f calories", f.calculateCalories());
    }
}
